src/System.Net.Ping/tests/FunctionalTests/UnixPingUtilityTests.cs

```diff
@@ -49,24 +49,48 @@ public class UnixPingUtilityTests
             int numPingsSent = ParseNumPingsSent(pingOutput);
             Assert.Equal(1, numPingsSent);
 
-            long rtt = UnixCommandLinePing.ParseRoundTripTime(pingOutput);
-            Assert.InRange(rtt, 0, long.MaxValue);
+            try
+            {
+                long rtt = UnixCommandLinePing.ParseRoundTripTime(pingOutput);
+                Assert.InRange(rtt, 0, long.MaxValue);
+            }
+            catch
+            {
+                Assert.Contains("should contain round trip time", pingOutput); // Will log the output
```

Comment:
Nit: I understand why you're doing this, but it seems like a hacky construct.  For situations like this, why not just replace these lines with something like:
```C#
catch (Exception e)
{
    throw new Exception("Message that provides more context", e);
}
```
?